### PR TITLE
Feature/initial setup

### PR DESCRIPTION
**Pull request by brianreichtcs**
If approved, this PR provides the following:

* Provides `php-unit-tests-yml`, GitHub Action Test Runner for PHPUnit tests
* Provides `phpcs.yml`, GitHub Action for validating changes against project coding standards
* Base `.gitignore`
* Stubbed-out project `README.md`
* Base `composer.json` that sets up project structure and basic dependencies
* Provides `phpcs.xml` which establishes project coding standards, based on PSR-12.
* Provides `phpunit.xml` which establishes unit test suite settings
* Provides `JunkTest.php` which exists to verify PHPUnit action runs correctly.

### CODE REVIEW

File: .github/workflows/phpcs.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Install PHP_CodeSniffer
+        run: |
+          curl -OL https://squizlabs.github.io/PHP_CodeSniffer/phpcs.phar
+          php phpcs.phar --version
+
+      - uses: tinovyatkin/action-php-codesniffer@v1
+        with:
+          files: "**.php" # you may customize glob as needed
+          phpcs_path: php phpcs.phar
+          standard: phpcs.xml
```
Comment (tomegantcs) [inline .github/workflows/phpcs.yml:37]: NIT Please add final newline to appease git

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,4 @@
 composer.phar
+composer.lock
 /vendor/
-
-# Commit your application's lock file https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control
-# You may choose to ignore a library lock file http://getcomposer.org/doc/02-libraries.md#lock-file
-# composer.lock
+.phpunit.result.cache
```
Comment (tomegantcs) [inline .gitignore:4]: NIT Please add final newline to appease git

File: phpcs.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <exclude name="PSR12.Traits.UseDeclaration.UseAfterBrace" />
+    </rule>
+    <rule ref="PSR2">
+        <exclude name="Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine" />
+        <exclude name="PSR2.Classes.ClassDeclaration.OpenBraceNewLine" />
+        <exclude name="PSR2.ControlStructures.ControlStructureSpacing" />
+    </rule>
+    <rule ref="Generic.Functions.OpeningFunctionBraceKernighanRitchie" />
+    <rule ref="Generic.Classes.OpeningBraceSameLine" />
+</ruleset>
```
Comment (tomegantcs) [inline phpcs.xml:30]: NIT Please add final newline to appease git

File: phpcs.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" name="PHP_CodeSniffer" xsi:noNamespaceSchemaLocation="phpcs.xsd">
+    <description>The coding standard for CORE.</description>
+
+    <file>src</file>
+    <file>tests</file>
+    <!-- Currently excluding everything else from PHPCS because why force the brain damage onto legacy code? -->
```
Comment (tomegantcs) [inline phpcs.xml:7]: NIT this line is inappropriate for this repository

File: .github/workflows/php-unit-tests.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    branches:
+      - 'master'
+      - 'develop'
+      - 'releases/**'
+  pull_request:
+
+jobs:
+  run-php-unit-tests:
+    name: Run PHP Unit Tests
+    runs-on: ubuntu-18.04
```
Comment (tomegantcs) [inline .github/workflows/php-unit-tests.yml:19]: Do we want to run on latest?

**Review by tomegantcs — APPROVED** [review]

File: .github/workflows/php-unit-tests.yml (hunk 5 above)
Comment (tomegantcs) [reply .github/workflows/php-unit-tests.yml]: I'm not sure what to do about this one honestly. I think 18.04 is fine for now if it works but feel like we need to plan some time to dig into this in a comprehensive way.

**Comment by brianreichtcs** [comment]
@tomegantcs  and @willsantanentcs I've made the tweaks to resolve Tom's nits. Also added license and a blurb about not being affiliated with SmarterU's parent company, which was requested by their lead dev.

File: .github/workflows/phpcs.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (tomegantcs) [inline .github/workflows/phpcs.yml:38]: I think you have some white space characters on this final line now which should be removed to appease git.

**Pull request merged**